pkg/obfuscator/keywords_test.go

```diff
@@ -9,34 +9,34 @@ import (
 func TestNewKeywordsObfuscator(t *testing.T) {
 	for _, tc := range []struct {
 		name           string
-		replacements   map[string]string
+		replacements   map[string]Replacement
 		input          string
 		expectedOutput string
 		expectLegend   map[string]string
 	}{
 		{
 			name: "basic",
-			replacements: map[string]string{
-				"unique-word": "replacement",
+			replacements: map[string]Replacement{
+				"unique-word": Replacement{ReplacedWith: "replacement"},
 			},
 			input:          "input with unique-word",
 			expectedOutput: "input with replacement",
 			expectLegend:   map[string]string{"unique-word": "replacement"},
 		},
 		{
 			name: "no replacement",
-			replacements: map[string]string{
-				"unique-word": "replacement",
+			replacements: map[string]Replacement{
+				"unique-word": Replacement{ReplacedWith: "replacement"},
 			},
 			input:          "input with common words",
 			expectedOutput: "input with common words",
 			expectLegend:   map[string]string{},
 		},
 		{
 			name: "partial replacement",
-			replacements: map[string]string{
-				"first-unique":  "first-replacement",
-				"second-unique": "second-replacement",
+			replacements: map[string]Replacement{
+				"first-unique":  Replacement{ReplacedWith: "first-replacement"},
+				"second-unique": Replacement{ReplacedWith: "second-replacement"},
 			},
 			input:          "input with first-unique word",
 			expectedOutput: "input with first-replacement word",
```

Comment:
I personally would've left the test and input intact, you just have to change the below Equal to o.Report().AsMap(). Should work the exact same way. No need to change all those tests